src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -558,10 +557,28 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
 
         return FieldValueInfo.newFieldValueInfo(LIST)
                 .fieldValue(overallResult)
-                .fieldValueInfos(fieldValueInfos)
+                .fieldValueInfos(combineFutures(fieldValueInfos))
                 .build();
     }
 
+    private <T> List<T> combineFutures(final List<CompletableFuture<T>> futures) {
+        return futures
+                .stream()
+                .map(CompletableFuture::join)
+                .collect(Collectors.toList());
+    }
+
+    private CompletableFuture<List<ExecutionResult>> getExecutionResults(
+            final List<CompletableFuture<FieldValueInfo>> fieldValueInfos) {
+        return CompletableFuture.supplyAsync(() ->
+                combineFutures(fieldValueInfos)
+                        .stream()
+                        .map(FieldValueInfo::getFieldValue)
+                        .map(CompletableFuture::join)
+                        .collect(Collectors.toList())
+        );
+    }
+
```

Comment:
Once again this introduces more threading (via `supplyAsync` and then synchs on them twice via 2 joins.